workbench/src/renderer/InvestJob.js

```diff
@@ -32,6 +32,19 @@ export default class InvestJob {
         (key) => investJobStore.getItem(key)
       ));
     }
+    // Migrate old-style jobs
+    // We can eventually remove this code once it's likely that most users
+    // will have updated and ran a newer version of the workbench
+    jobArray = await Promise.all(jobArray.map(async (job) => {
```

Comment:
It seems like we could condense this mapping into the `Promise.all(sortedJobHashes.map(` from above and then only map over the jobs one time?

Reply:
We could, I had kept it separate to make it easier to remove in the future, but it would be worth condensing if you think it's a performance concern?